artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -2733,7 +2799,7 @@ private boolean deliver() {
    }
 
    protected void removeMessageReference(ConsumerHolder<? extends Consumer> holder, MessageReference ref) {
-      if (!nonDestructive) {
+      if (!isNonDestructive()) {
```

Comment:
why the change here? again its internal to class, and hot path, thus was field access

Reply:
(nit)